File: artifacts/api-server/src/sql-agent/state.ts
import { TableDefinition } from "./table-registry.js";

export interface AgentState {
  userQuery: string;
  iteration: number;
  maxIterations: number;
  relevantTables: TableDefinition[];
  schemaContext: string;
  generatedSQL: string;
  validationPassed: boolean;
  validationFeedback: string;
  finalSQL: string;
  error: string;
  agentLog: string[];
}

export const initialState = (
  userQuery: string,
  maxIterations = 5
): AgentState => ({
  userQuery,
  iteration: 0,
  maxIterations,
  relevantTables: [],
  schemaContext: "",
  generatedSQL: "",
  validationPassed: false,
  validationFeedback: "",
  finalSQL: "",
  error: "",
  agentLog: [],
});

File: artifacts/fastapi-server/state.py
from dataclasses import dataclass, field
from typing import List, Any


@dataclass
class AgentState:
    user_query: str
    iteration: int = 0
    max_iterations: int = 5
    relevant_tables: List[Any] = field(default_factory=list)
    schema_context: str = ""
    generated_sql: str = ""
    validation_passed: bool = False
    validation_feedback: str = ""
    final_sql: str = ""
    error: str = ""
    agent_log: List[str] = field(default_factory=list)
